llms/mlx_lm/tuner/trainer.py

```diff
@@ -258,7 +263,7 @@ def step(batch):
                     "val_loss": val_loss,
                     "val_time": val_time,
                 }
-                training_callback.on_val_loss_report(val_info)
+                training_callback.on_val_loss_report(val_info, args=args)
```

Comment:
Similarly it should be easy enough to enclose the `args` in the user provided loss reporting functions.